torch/distributions/transforms.py

```diff
@@ -125,7 +125,8 @@ def inv(self):
             inv = self._inv()
         if inv is None:
             inv = _InverseTransform(self)
-            self._inv = weakref.ref(inv)
+            if not torch.compiler.is_dynamo_compiling():
+                self._inv = weakref.ref(inv)
```

Comment:
If dynamo not support `weakref`s, I would guess the workaround would be to replace each `weakref` with some sort of api compatible `strongref` and just accept that the resulting reference cycles will need a gc pass to free memory, rather than `weakref` which needs only refcounts.

Reply:
I think compile should support weakrefs but I wonder if the issue isn't that they're supported as long as the object has been created outside of the compiled region, not within.